server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/ConsistencyTasksIntegrationTest.java

```diff
@@ -312,7 +334,50 @@ void shouldRecomputeQuotas(GuiceJamesServer server) throws Exception {
     }
 
     @Test
-    void solveCassandraMappingInconsistencyShouldSolveNotingWhenNoInconsistencies() {
+    void shouldSolveMessagesInconsistency(GuiceJamesServer server) {
+        MailboxProbeImpl probe = server.getProbe(MailboxProbeImpl.class);
+        MailboxPath inbox = MailboxPath.inbox(BOB);
+        MailboxId mailboxId = probe.createMailbox(inbox);
+
+        TestingSessionProbe testingProbe = server.getProbe(TestingSessionProbe.class);
+        testingProbe.getTestingSession().registerScenario(fail()
+            .times(6) // Insertion in the DAO is retried 5 times once it failed
+            .whenQueryStartsWith("INSERT INTO messageIdTable"));
+
+        try {
+            probe.appendMessage(BOB.asString(), inbox,
+                new ByteArrayInputStream(MESSAGE_CONTENT_AS_BYTES), DATE,
+                !IS_RECENT, FLAGS);
```

Comment:
Can we use `public ComposedMessageId appendMessage(String username, MailboxPath mailboxPath, MessageManager.AppendCommand appendCommand)` instead?